src/bun.js/api/bun/subprocess.zig

```diff
@@ -1802,6 +1845,41 @@ pub fn getExited(
     }
 }
 
+pub fn getSuccess(
+    this: *Subprocess,
+    this_value: JSValue,
+    globalThis: *JSGlobalObject,
+) JSValue {
+    if (JSC.Codegen.JSSubprocess.successPromiseGetCached(this_value)) |promise| {
+        return promise;
+    }
+
+    switch (this.process.status) {
+        .exited => |exit| {
+            if (exit.code != 0) {
+                // TODO: this should be a real `ShellError`?
```

Comment:
I was unable to figure out how to construct an actual `ShellError`, since that's defined in TypeScript. I'm assuming I need to get it from `globalThis`, and could also use advice on this.

Reply:
I'm not sure what the error instance should be here yet anyway. Since this is not really a Bun shell API my gut feeling is that ShellError is not relevant.

Perhaps we would just need a new class "SubprocessFailureError" which contains stdout/stderr properties as buffers